### PR TITLE
unblock gc fix for read only files

### PR DESCRIPTION
**Pull request by NaraVen**

### CODE REVIEW

File: blobfuse/src/BlobfuseGcCache.cpp [hunk 1]
```diff
@@ -164,6 +164,19 @@ void gc_cache::run_gc_cache()
             {
                 //clean up the file from cache
                 int fd = open(mntPath, O_WRONLY);
+                if (errno == 13) {
+                    close(fd);
```
Comment (vibhansa-msft) [inline blobfuse/src/BlobfuseGcCache.cpp:168]: We can add a log here stating this condition has hit and file is write protected. If the chmod fails we will not know whether this code flow was hit or not

File: blobfuse/src/BlobfuseGcCache.cpp [hunk 2]
```diff
@@ -164,6 +164,19 @@ void gc_cache::run_gc_cache()
             {
                 //clean up the file from cache
                 int fd = open(mntPath, O_WRONLY);
+                if (errno == 13) {
```
Comment (vibhansa-msft) [inline blobfuse/src/BlobfuseGcCache.cpp:167]: errno = 13 : then open was not successful so there is no need to close

File: blobfuse/src/BlobfuseGcCache.cpp [hunk 3]
```diff
@@ -164,6 +164,19 @@ void gc_cache::run_gc_cache()
...
+                    close(fd);
+                    int fdr = open(mntPath, O_RDONLY);
+                    if (fdr > 0) {
+                        errno = 0;
+                        if (fchmod(fdr, config_options.defaultPermission) == 0) {
+                            AZS_DEBUGLOGV("GCClean clould not open for delete, so, Write protection overwritten for %s, resubmitting for cleanup", mntPath);
+                            }
+                        //TODO: If there is lock issue below convert permissions back to what it was.
+                        close(fdr);
+                    }
```
Comment (vibhansa-msft) [inline blobfuse/src/BlobfuseGcCache.cpp:177]: If readonly also fails then we can dump a log saying this file will not get evicted from cache due to the error.

File: blobfuse/src/BlobfuseGcCache.cpp [hunk 4]
```diff
@@ -164,6 +164,19 @@ void gc_cache::run_gc_cache()
             {
                 //clean up the file from cache
                 int fd = open(mntPath, O_WRONLY);
+                if (errno == 13) {
+                    close(fd);
+                    int fdr = open(mntPath, O_RDONLY);
+                    if (fdr > 0) {
+                        errno = 0;
+                        if (fchmod(fdr, config_options.defaultPermission) == 0) {
```
Comment (vibhansa-msft) [inline blobfuse/src/BlobfuseGcCache.cpp:172]: chmod failure can also be logged here

Comment (NaraVen) [reply blobfuse/src/BlobfuseGcCache.cpp:176]: done

File: blobfuse/src/BlobfuseGcCache.cpp (hunk 3 above)
Comment (NaraVen) [reply blobfuse/src/BlobfuseGcCache.cpp:186]: done

File: blobfuse/src/BlobfuseGcCache.cpp (hunk 2 above)
Comment (NaraVen) [reply blobfuse/src/BlobfuseGcCache.cpp:170]: done

Comment (NaraVen) [reply blobfuse/src/BlobfuseGcCache.cpp:170]: done

File: blobfuse/src/BlobfuseGcCache.cpp (hunk 1 above)
Comment (NaraVen) [reply blobfuse/src/BlobfuseGcCache.cpp]: done

**Review by vibhansa-msft — APPROVED** [review]

**Pull request merged**